cmd/ci-operator/main.go

```diff
@@ -419,6 +419,10 @@ func (o *options) Complete() error {
 			return fmt.Errorf("failed to generate secret %s: %v", name, err)
 		}
 		secret.Name = name
+		if len(o.variant) > 0 {
+			secret.Name = fmt.Sprintf("%s-%s", o.variant, name)
+		}
```

Comment:
This does not seem to be a correct thing to do - the `o.variant` is only used to specify which config to obtain from the configresolver, otherwise ci-operator has no concept of a "variant" internally, and I think we should keep it that way.

If something that ci-operator creates needs to match to whatever `JOB_NAME_SAFE` is, let's base its name directly on `JOB_NAME_SAFE`, not on something that affects it.

Reply:
in this case, we don't need to update `ci-operator` at all but instead `prowgen` should add the correct secret name including the variant in the `--secret-dir` flag